### PR TITLE
Server start skript

### PR DESCRIPTION
**Pull request by formtapez**

### CODE REVIEW

File: tools/scripts/server/startServer.bat [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
-echo Trying to copy config file. This might take a while...
-:copyLoop
-	copy "%~dp0.\.\serverConfig.cfg" "%armaDir%" > NUL 2>&1
-
-	if not [%errorlevel%] == [0]  (
-		:: sleep 100ms
-		pathping localhost -n -q 1 -p 100 >nul
-		goto :copyLoop
-	)
+copy /Y "%~dp0.\serverConfig.cfg" "%armaDir%" > NUL 2>&1
```
Comment (formtapez) [inline tools/scripts/server/startServer.bat]: This should never fail. If so, abort! No need to retry on and on...

Comment (Krzmbrzl) [reply tools/scripts/server/startServer.bat]: It fails when you are restarting the server because windows keeps the file locked some time even after the server-instance has been killed. And in these cases the copying fails

File: tools/scripts/server/stopServer.bat [hunk 2]
```diff
@@ -19,13 +19,13 @@ echo Killing the (potentially) running server. This might take a while...
 	taskkill /im %exeName% > NUL 2>&1
 	
 	if [%errorlevel%] == [0] (
-		:: sleep 100ms
-		pathping localhost -n -q 1 -p 100 >nul
+		:: sleep 1s
```
Comment (Krzmbrzl) [inline tools/scripts/server/stopServer.bat]: Waiting a full second in between every iteration is a pain in the *** and can increase the wait-time significantly. Why did you change a working solution?

File: tools/scripts/server/setMetaData.bat.example [hunk 3]
```diff
@@ -4,4 +4,3 @@
 :: Set the meta information needed for the start/stop-server scripts
 set "armaDir=C:\Program Files (x86)\Steam\steamapps\common\Arma 3"
 set "exeName=arma3server_x64.exe"
-set "modDir=C:\Program Files (x86)\Steam\steamapps\common\Arma 3"
```
Comment (Krzmbrzl) [inline tools/scripts/server/setMetaData.bat.example:7]: Why did you remove the mod-Dir? For instance on my machine this is different from the armaDir

File: tools/scripts/server/startServer.bat [hunk 4]
```diff
@@ -1,30 +1,36 @@
...
+if not [%errorlevel%] == [0]  (
+	echo Unable to copy server-config. Press any key to exit...
+	pause > NUL
+	exit 1
```
Comment (Krzmbrzl) [inline tools/scripts/server/startServer.bat]: With this solution the script fails instantly whenever windows has the respective file locked for just  another half a second. I don't think this is a convenient solution

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]

File: tools/scripts/server/stopServer.bat (hunk 2 above)
Comment (formtapez) [reply tools/scripts/server/stopServer.bat]: I removed pathping because it relies on specific firewall and DNS configuration which is a bit overkill to just wait a bit, and can fail catastsrophically when your setup differs. We can use windows' native sleep function "TIMEOUT". Unfortunately the smallest delay of it is 1 second, but that should not be a problem, because it only delays when an error occures - which never happened to me. Maybe my hard-kill modification in stopServer.bat helped to get rid of the file-lock.

File: tools/scripts/server/setMetaData.bat.example (hunk 3 above)
Comment (formtapez) [reply tools/scripts/server/setMetaData.bat.example:7]: I removed the mod-dir because it is not used anywhere. No mods are copied anymore.

Comment (Krzmbrzl) [reply tools/scripts/server/setMetaData.bat.example]: Ah I kinda missed that you changed the killing as well... Though I'm not quite sure if the arma server responds well to hard-killing... Iirc I read something about this causing problems if restarting the server immediately after :thinking:

File: tools/scripts/server/stopServer.bat (hunk 2 above)
Comment (Krzmbrzl) [reply tools/scripts/server/stopServer.bat]: Ah I kinda missed that you changed the killing as well... Though I'm not quite sure if the arma server responds well to hard-killing... Iirc I read something about this causing problems if restarting the server immediately after :thinking:

File: tools/scripts/server/setMetaData.bat.example (hunk 3 above)
Comment (Krzmbrzl) [reply tools/scripts/server/setMetaData.bat.example:7]: So how are the mods supposed to get where they are needed?

Comment (formtapez) [reply tools/scripts/server/setMetaData.bat.example:7]: The mods stay where they were build, e.g. in `PBOs\dev\@OPT`
Armaserver can load them from there - its not that limited like the location of the serverconfig-file.

Comment (Krzmbrzl) [reply tools/scripts/server/setMetaData.bat.example:7]: Really? I thought I had tested this before and it didn't work. Apparently I need to retry then though :man_shrugging:

Comment (formtapez) [reply tools/scripts/server/setMetaData.bat.example:7]: Yeah, but it only works with absolute paths, thats why i introduced that conversion.

File: tools/scripts/server/startServer.bat [hunk 5]
```diff
@@ -1,30 +1,37 @@
...
+		TIMEOUT /NOBREAK /T 1 > NUL
 		goto :copyLoop
 	)
 
 :: Start the server and minimize it once it's started
-start /min %exeName% -config=serverConfig.cfg -serverMod="%clib%;%opt%" -debugCallExtension
-
-echo Successfully started the server
\ No newline at end of file
+start /min %exeName% -config=serverConfig.cfg -serverMod="%clib%;%opt%;%additionalMods%" -debugCallExtension
```
Comment (Krzmbrzl) [inline tools/scripts/server/startServer.bat]: `%additionalMods%`  aren't serverMods. They should be passed using the `-mod` parameter

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]

File: tools/scripts/server/startServer.bat [hunk 6]
```diff
@@ -1,30 +1,37 @@
...
 :: is being called from this one
 @echo off
 
 :: set meta infos
-call %~dp0.\setMetaData.bat
+call "%~dp0.\setMetaData.bat"
+
+:: convert relative to absolute path of clib
+:: so armaserver can read it properly
+pushd "%~dp0..\..\..\PBOs\dev\@CLib"
```
Comment (Krzmbrzl) [inline tools/scripts/server/startServer.bat:16]: Before this line there should be a check whether the directory actually exists. If not, exit with a useful error message (like "Can't find mod-Dir - did you forget to build the mods before?")

File: tools/scripts/server/startServer.bat [hunk 7]
```diff
@@ -1,30 +1,37 @@
...
 
-set "clib=%modDir%\@CLib"
-set "opt=%modDir%\@OPT"
-
 :: Copy the server-config into the arma-dir as Arma can only read configs relative
 :: to the exe
-echo Trying to copy config file. This might take a while...
 :copyLoop
-	copy "%~dp0.\.\serverConfig.cfg" "%armaDir%" > NUL 2>&1
+	echo Trying to copy config file. This might take a while...
```
Comment (Krzmbrzl) [inline tools/scripts/server/startServer.bat]: The echo should be outside the loop (before the `:copyLoop`)

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]

File: tools/scripts/server/stopServer.bat (hunk 2 above)
Comment (Krzmbrzl) [reply tools/scripts/server/stopServer.bat]: So I just tested it on my machine and the looping is absolutely required in order for the server to actually getting killed (when restarting it).
Thus waiting a full second in between the iterations isn't really acceptable...

Comment (Krzmbrzl) [reply tools/scripts/server/stopServer.bat]: I think I found a solution: https://stackoverflow.com/a/29879492/3907364 (second option)
This uses JScript to do the waiting which is way more powerful than batch's timeout. Simply copy the provided code (and a link to that SO-question) in a batch-file called `sleep.bat` and then inside our batch code we write `call %~dp0.\sleep.bat 100` to wait 100 ms

Comment (formtapez) [reply tools/scripts/server/stopServer.bat]: I watched that stopServer routine on my machine again as well. It takes 2-3 iterations (2-3 seconds) to get it (soft)-killed.
When we go back to your 100 ms delay, we would have 20 - 30 iterations and would statistically be able to restart the server 0.4 seconds earlier.
By looking at your SO-example i came to the conclusion that this 0.4 seconds are not worth it to include such a piece of garbage code :-) Then i'd rather go back to your traceping method or include some solid linux tools like "sleep" as .exe file. (We'll probably need more of them in the future anyway - sed, awk, cut, ...)

Comment (Krzmbrzl) [reply tools/scripts/server/stopServer.bat]: Well using Linux tools is not really an option, is it? 'cause after all this has to run on windows... 
And I don't se what's wrong with the SO code. Noone has to look at it once it is there - and it does what it's asked for ^^

Comment (formtapez) [reply tools/scripts/server/stopServer.bat]: ok i was looking at a different script from that page.
its integrated now - please merge this commits so i can delete the repo to fix my git problem.

**Review by Krzmbrzl — APPROVED** [review]
